packages/x-date-pickers/src/internals/hooks/useField/useFieldV6TextField.ts

```diff
@@ -196,6 +197,7 @@ export const useFieldV6TextField: UseFieldTextField<false> = (params) => {
 
   const syncSelectionFromDOM = () => {
     if (readOnly) {
+      setSelectedSections(null);
```

Comment:
Revert https://github.com/mui/mui-x/pull/12593/files#diff-c8baae3f915827c7c3a382b1b2e21197c74844d930c08a95a68f39f9e277da77L192 in favor of a more isolated fix above.